lib/spack/spack/util/web.py

```diff
@@ -36,7 +43,12 @@ class HTMLParseError(Exception):
 import spack.stage
 import spack.error
 import spack.util.crypto
+
 from spack.util.compression import ALLOWED_ARCHIVE_TYPES
+from spack.util.s3 import create_s3_session
+from spack.util.url import join as urljoin, parse as urlparse
+
+from spack.s3_handler import open as s3_open
```

Comment:
We should rethink the module breakdown here, as there are a lot of circular imports.  In general we want to avoid any circular dependencies between module in Spack (I know there are already quite a few)